### PR TITLE
Delete user's posts & signups directly.

### PR DESCRIPTION
**Pull request by DFurnes**
### What's this PR do?

This pull request updates our `UserObserver` to directly delete signups & posts when a user is deleted, rather than handing that task off to Rogue via its [`DELETE users/{user}`](https://github.com/DoSomething/rogue/blob/4d43ca518ab629af0dd0297a772615f8a824350a/app/Http/Controllers/UsersController.php#L26-L71) endpoint. To ensure we always properly anonymize deleted posts, I've opted to handle this automatically via observers.

### How should this be reviewed?

👁️ 🗑️ 

### Any background context you want to provide?

Unlike our other PRs, this will take effect immediately once we deploy it!! 😱 So we'll definitely want to keep an eye on the first round of account deletions after this is deployed to ensure that everything continues to function as expected.

### Relevant tickets

References [Pivotal #176851322](https://www.pivotaltracker.com/story/show/176851322).

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [ ] Added appropriate feature/unit tests.
- [ ] If new attributes were added to users, there is a corresponding PR to surface these in Aurora.
- [ ] If new attributes were added to users, then the data team already knows about these changes.

### CODE REVIEW

File: app/Jobs/DeleteUserFromOtherServices.php [hunk 1]
```diff
@@ -41,7 +41,6 @@ public function handle()
             ->then(function () {
                 app(CustomerIo::class)->deleteUser($this->id);
                 app(Gambit::class)->deleteUser($this->id);
-                app(Rogue::class)->deleteUser($this->id);
```
Comment (weerd) [inline app/Jobs/DeleteUserFromOtherServices.php:44]: 👋

File: tests/CreatesApplication.php [hunk 2]
```diff
@@ -46,6 +46,13 @@ public function createApplication()
...
+        $this->customerIoMock->shouldReceive('deleteUser');
+
+        $this->fastlyMock = $this->mock(\App\Services\Fastly::class);
+        $this->fastlyMock->shouldReceive('purge');
+
+        $this->gambitMock = $this->mock(\App\Services\Gambit::class);
+        $this->gambitMock->shouldReceive('deleteUser');
 
         $this->fastlyMock = $this->mock(\App\Services\Fastly::class);
         $this->fastlyMock->shouldReceive('purge');
```
Comment (weerd) [inline tests/CreatesApplication.php:58]: The `fastlyMock` sequence shouldn't be there twice right?

File: tests/TestCase.php [hunk 3]
```diff
@@ -154,12 +155,34 @@ public function registerUpdated()
...
     }
 
+    /**
+     * Assert the given models are soft-deleted and the provided
+     * fields have been nulled out.
+     *
+     * @param Collection $collection
+     * @param array $fields
+     */
+    public function assertAnonymized(Collection $collection, array $fields = [])
```
Comment (weerd) [inline tests/TestCase.php:165]: Nice! Very useful 👏

**Review by weerd — APPROVED** [review]
🎉  Looks good to me! Just a quick ques on a potential duplication!

File: tests/CreatesApplication.php (hunk 2 above)
Comment (DFurnes) [reply tests/CreatesApplication.php:58]: Oh yep, typo!

**Comment by DFurnes** [comment]
Rebased to fix merge conflicts, then fixed up some failing tests in [`436e5da...08fefd2`](https://github.com/DoSomething/northstar/pull/1122/files/33725c9137dc3eabd3408102f397bb183a309bd5..08fefd2abd68c615b40b3a37a0483d18ddd321d1).

**Comment by DFurnes** [comment]
Finally, removed the Rogue API client in 8b9c019! 🍃

**Comment by DFurnes** [comment]
@weerd This should be ready for merge now! I've linked the changes since last review in the last two comments!

**Review by weerd — APPROVED** [review]
🎉

**Pull request merged**